control.lua

```diff
@@ -872,11 +873,12 @@ script.on_event({defines.events.on_tick}, function(event)
 			end
 		end
 		-- Add disabled prototypes
-		if #noxy_trees.disabled_match > 0 then
+		if next(noxy_trees.disabled_match) then
 			for e,_ in pairs(game.entity_prototypes) do
-				for k,_ in pairs(noxy_trees.disabled_match) do
-					if e.find(k) then
-						noxy_trees.disabled[e] = true
+				for k,v in pairs(noxy_trees.disabled_match) do
+					if e:find(k) then
+						noxy_trees.disabled[e] = v
```

Comment:
Please leave this line as it was. The extra value lookup is not necessary here.

Reply:
As the noxy_trees.disabled_match is array of true or false, I though it makes sense to use the value here, please reconsider?

Reply:
The array will never have a value of false.